### PR TITLE
Use publish-image action

### PR DESCRIPTION
**Pull request by brooksn**
#### Proposed Changes ####

- Updates the release workflow to use the publish-image action to run image/manifest push scripts
- Pushes images directly to the prime registry
- Moves the windows image push into a separate step
- Adds SBOM to image manifests

#### Linked Issues ####

https://github.com/rancher/ecm-distro-tools/issues/481

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -57,12 +60,37 @@ jobs:
...
-        DOCKER_PASSWORD: ${{ env.DOCKER_PASSWORD }}
-    
+      uses: rancher/ecm-distro-tools/actions/publish-image@master
+      with:
+        image: "rke2-runtime"
+        tag: ${{ github.ref_name }}
+        make-target: publish-image-runtime
+
+        public-repo: rancher
+        public-username: ${{ env.DOCKER_USERNAME }}
```
Comment (nicholasSUSE) [inline .github/workflows/release.yml:70]: Should we use secrets for all the sensitive data? 
```(yaml)
          public-repo: rancher
          public-username: ${{ secrets.DOCKER_USERNAME }}
          public-password: ${{ secrets.DOCKER_PASSWORD }}
          prime-repo: rancher
          prime-registry: ${{ secrets.PRIME_REGISTRY }}
          prime-username: ${{ secrets.PRIME_REGISTRY_USERNAME }}
          prime-password: ${{ secrets.PRIME_REGISTRY_PASSWORD }}
```

**Review by nicholasSUSE — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (rafaelbreno) [reply .github/workflows/release.yml:70]: We're getting those values from vault, the `"Read secrets"` step will retrieve them from vault and set it as `envs`. The Github CI will mask those, so no secrets leak.

**Review by nicholasSUSE — APPROVED** [review]

File: scripts/publish-manifest-runtime [hunk 2]
```diff
@@ -16,8 +16,10 @@ docker manifest annotate ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION} ${REPO}/$
 docker manifest annotate ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION} ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION}-linux-arm64 --os linux --arch arm64
 docker manifest annotate ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION} ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION}-windows-amd64 --os windows --arch amd64
 
-set +x
-docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
-set -x
-
 docker manifest push ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION}
+
+if [ -n "$IID_FILE_FLAG" ]; then
```
Comment (briandowns) [inline scripts/publish-manifest-runtime:21]: Please stay consistent with the use of parenthesis and curly braces for make and shell variables.

File: scripts/build-image-runtime [hunk 3]
```diff
@@ -5,7 +5,13 @@ cd $(dirname $0)/..
 
 source ./scripts/version.sh
 
+./scripts/setup-docker-builder
+
 DOCKER_BUILDKIT=${DOCKER_BUILDKIT:-1} docker image build \
+    --sbom=true \
+    --builder ${PROG} \
+    ${IID_FILE_FLAG} \
```
Comment (brandond) [inline scripts/build-image-runtime:13]: What is this flag, and where does the env var setting it come from?

**Pull request closed without merging**